### PR TITLE
Progress callbacks now pass in an accuracy parameter, and start at 0

### PR DESCRIPTION
**Pull request by jstefaniuk-d2l**

### CODE REVIEW

File: src/plugins/generic/viewer.js [hunk 1]
```diff
@@ -12,7 +12,13 @@ var Viewer = React.createClass({
 		filename: React.PropTypes.string.isRequired,
 		size: React.PropTypes.number.isRequired,
 		srcdownload: React.PropTypes.string,
-		locale: React.PropTypes.string
+		locale: React.PropTypes.string,
+		progressCallback: React.PropTypes.func
+	},
+	componentWillMount: function() {
```
Comment (CodeBaboon) [inline src/plugins/generic/viewer.js]: should be componentDidMount I think (not that it should really have an impact)

File: src/plugins/image/__tests__/viewerTests.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ describe('Image Viewer', function() {
 		expect(React.findDOMNode(img).src).toBe('foo.bar');
 	});
 
-	it('Calls the progressCallback and passes 100 in as the value', function() {
+	it('Calls the progressCallback twice and passes 0, than 100 in as the value', function() {
```
Comment (CodeBaboon) [inline src/plugins/image/__tests__/viewerTests.js]: then not than (#grammar #bikeshed)

Comment (jstefaniuk-d2l) [inline src/plugins/image/__tests__/viewerTests.js]: whoops, my bad

Comment (CodeBaboon) [inline src/plugins/image/__tests__/viewerTests.js]: This feels like 4 different tests to me.

- Calls the progressCallback twice
- First call is 0
- Second call is 100
- Calls have certainty none

If we don't want to repeat the setup each time we could probably create a new test file specific to progress callback testing that does it in a beforeEach().

File: src/plugins/pdf/native/nativeViewer.js [hunk 3]
```diff
@@ -4,12 +4,15 @@ var React = require('react'),
...
+	componentWillMount: function() {
+		this.updateProgress(0);
+	},
 	componentDidMount: function() {
 		this.updateProgress(100);
 	},
 	updateProgress: function(progress) {
 		if (this.props.progressCallback) {
-			this.props.progressCallback(progress);
+			this.props.progressCallback(progress, 'none');
```
Comment (CodeBaboon) [inline src/plugins/pdf/native/nativeViewer.js]: I'm not seeing this for alternateViewer, probably need to rebase off latest to get that and then add this change and tests?

File: README.md [hunk 4]
```diff
@@ -73,7 +73,13 @@ Then open `http://localhost:8080/` in your browser.
 ## Parameters
 FileViewer has the following parameters:
 
-`progressCallback` Takes a callback function, which should accept a decimal, indicating the percentage loaded out of 100. May or may not be accurate.
+`progressCallback` Takes a callback function, which should accept 2 parameters, the first reflecting the progress, and the second reflecting the accuracy of the progress value.
+
+* progress: A number between 0 and 100 indicating the percentage loaded.
+* accuracy: The accuracy of the returned progress value, currently we choose between two options:
+	* `certain`: indicates that progress will be a number between 0 and 100
+	* `guessing`: indicates that progress will be a number between 0 and 100, and that the progress may not be accurate
```
Comment (d2l-joelp) [inline README.md]: Nitpicky, but I would probably rename this to just 'guess'.

Comment (jstefaniuk-d2l) [inline README.md]: good call

File: src/plugins/image/__tests__/viewerTests.js [hunk 5]
```diff
@@ -51,7 +51,46 @@ describe('Image Viewer', function() {
 				progressCallback={progressFunc} />
 		);
 
-		expect(progressFunc.mock.calls.length).toBe(1);
-		expect(progressFunc.mock.calls[0][0]).toBe(100);
+		expect(progressFunc.mock.calls.length).toBe(2);
```
Comment (d2l-joelp) [inline src/plugins/image/__tests__/viewerTests.js]: This test seems redundant given the other three tests. If you want to ensure it is called exactly twice you can just move this expect into the test 'Calls the progressCallback both times with certainty none' and maybe change 'both times' to twice.

File: src/plugins/pdf/native/__tests__/nativeViewerTests.js [hunk 6]
```diff
@@ -28,7 +28,46 @@ describe('PDF Native Viewer', function() {
 				progressCallback={progressFunc} />
 		);
 
-		expect(progressFunc.mock.calls.length).toBe(1);
-		expect(progressFunc.mock.calls[0][0]).toBe(100);
+		expect(progressFunc.mock.calls.length).toBe(2);
```
Comment (d2l-joelp) [inline src/plugins/pdf/native/__tests__/nativeViewerTests.js]: Same as comment for image viewer tests.

**Comment by CodeBaboon** [comment]
LGTM

**Pull request merged**